tools/SourceKit/CMakeLists.txt

```diff
@@ -138,20 +143,12 @@ if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
   set_target_properties(dispatch
                         PROPERTIES
                           IMPORTED_LOCATION
-                            ${SWIFT_PATH_TO_LIBDISPATCH_BUILD}/src/${CMAKE_SHARED_LIBRARY_PREFIX}dispatch${CMAKE_SHARED_LIBRARY_SUFFIX}
+                            ${install_dir}/lib/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}dispatch${CMAKE_SHARED_LIBRARY_SUFFIX}
                           INTERFACE_INCLUDE_DIRECTORIES
                             ${install_dir}/include
                           IMPORTED_LINK_INTERFACE_LIBRARIES
                             swiftCore-${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}-${SWIFT_HOST_VARIANT_ARCH})
 
-  add_library(BlocksRuntime STATIC IMPORTED)
```

Comment:
The removal of `BlocksRuntime` is actually problematic for Windows builds, since we need that to enable SourceKit builds as well.  Ideally, we would use a shared BlocksRuntime, but, that can be handled separately.

Reply:
I still can't test on Windows, but at least on Linux the implementation of BlocksRuntime seems to live in libdispatch, so SourceKit should link that:

```
$ readelf --dyn-syms -W swift-4.2-RELEASE-ubuntu14.04/usr/lib/swift/linux/libdispatch.so | grep -E '(Block_copy|Block_release)'
   605: 0000000000060220   196 FUNC    GLOBAL DEFAULT   11 _Block_release
   740: 00000000000600f0     7 FUNC    GLOBAL DEFAULT   11 _Block_copy
   853: 0000000000060340   201 FUNC    GLOBAL DEFAULT   11 _Block_copy_collectable
```

Reply:
This is a bug :-)  It shouldn't be exposed from libdispatch.  The static linkage shouldnt expose the symbols (which is part of the problem) - you cant actually use blocks without the shared runtime as John and I discussed on https://github.com/apple/swift-corelibs-libdispatch/pull/362.  I think that I'm going to try to get a couple of my patches up and then look into that again.

Reply:
I'm not sure you can - for the simple reason that you can call `Block_copy` and `Block_release` from outside of dispatch on objects you pass to dispatch. @rjmccall  's statement in 
https://github.com/apple/swift-corelibs-libdispatch/pull/362#issuecomment-400039465 seems a far cry from "yes, it definitely works". AIUI, at best it might work if the implementations the program and the dispatch library can see are ABI compatible, but the best way to ensure that IMO is just to use the same definition in the first place.

Reply:
Right, the same definition would be through a shared library (which is what the PR was working towards).  I'm just saying that in the mean time, statically linking against it should get you unblocked right?